backends/gstreamer/datachannel.rs

```diff
@@ -129,15 +111,11 @@ impl WebRtcDataChannelBackend for GStreamerWebRtcDataChannel {
     }
 
     fn send(&self, message: &str) -> WebRtcResult {
-        // glib::object::SendWeakRef needs to be upgraded from the thread
-        // where it was created. In this case, the channel weak ref
-        // was created on the webrtc controller's thread.
-        self.webrtc_thread
-            .internal_event(InternalEvent::SendDataChannelMessage(
-                Box::new(self.channel.clone()),
-                message.to_owned(),
-            ));
-        Ok(())
+        self.channel
+            .0
+            .emit("send-string", &[&Value::from(message)])
```

Comment:
While at it, you could also hide the signal emission things in functions inside `impl DataChannel` :)